### PR TITLE
feat: support building on Android

### PR DESCRIPTION
**Pull request by LucasXu0**
### Requirement
- Install Android Studio
- Install SDK Tools
  - NDK 24.0.8215888
  - Android SDK Command
  - CMake 3.22.1
Screenshot 2023-10-17 at 17 32 00
Screenshot 2023-10-17 at 17 31 50

- Install cargo-ndk

#### Linux
Q:   **/usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found**
```
error: failed to run custom build command for `librocksdb-sys v0.11.0+8.1.1`

Caused by:
  process didn't exit successfully: `/home/runner/work/AppFlowy/AppFlowy/frontend/rust-lib/target/debug/build/librocksdb-sys-9c6d5c10f08cbb0c/build-script-build` (exit status: 101)
  --- stderr
  /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found
  thread 'main' panicked at 'unable to generate rocksdb bindings: ClangDiagnostic("/usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found\n")', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librocksdb-sys-0.11.0+8.1.1/build.rs:40:10
  stack backtrace:
```
A: **`sudo apt-get install gcc-multilib`**

### Preview
Screenshot 2023-10-16 at 22 50 55

### CODE REVIEW

**Comment by LucasXu0** [comment]
Hi, @rileyhawk1417. I created a new PR based on [yours](https://github.com/AppFlowy-IO/AppFlowy/pull/3184). PTAL.

File: frontend/scripts/makefile/mobile.toml [hunk 1]
```diff
@@ -33,6 +33,29 @@ script = [
...
+] }
+
+[tasks.sdk-build-android]
+private = true
+script = [
+  """
+    cd rust-lib/
+    rustup show
+    echo "cargo ndk -t arm64-v8a -o ./jniLibs build --features "${FLUTTER_DESKTOP_FEATURES}""
+    cargo ndk -t arm64-v8a -o ./jniLibs build --features "${FLUTTER_DESKTOP_FEATURES}"
```
Comment (rileyhawk1417) [inline frontend/scripts/makefile/mobile.toml:53]: This line could fail for other builds like `x86` for emulators. The solution would be to force specify the platform-api version:
`cargo ndk -t arm64-v8a -p 24 -o ./jniLibs build --features "${FLUTTER_DESKTOP_FEATURES}`
This way even if someone wants to build the backend for `armeabi-7va` they just have to change the target.

Comment (LucasXu0) [reply frontend/scripts/makefile/mobile.toml:53]: Yep. I have included `x86_64` in it.

Comment (LucasXu0) [reply frontend/scripts/makefile/mobile.toml:53]: Yep. I have included x86_64 in it.

**Pull request merged**